pyshelf/artifact_manager.py

```diff
@@ -14,22 +14,40 @@ def get_artifact(self, path):
                 pyshelf.cloud.StreamIterator|None
         """
         content = None
+        artifact_list = self.assign_artifact_links(path)
+
+        if not artifact_list:
+            with self.container.create_bucket_storage() as storage:
+                content = storage.get_artifact(path)
+
+        return content
+
+    def assign_artifact_links(self, path):
+        """
+            Assigns links for an artifact or collection or artifacts.
+
+            Args:
+                path(string): path or name of artifact.
+
+            Returns:
+                List[string]: List of artifacts for collection of artifacts.
+        """
         with self.container.create_bucket_storage() as storage:
             if path[-1] != "/":
                 directory_path = path + "/"
             else:
                 directory_path = path
 
             artifact_list = storage.get_directory_contents(directory_path, recursive=False)
-            if len(artifact_list) > 0:
+
+            if artifact_list:
                 self.container.logger.debug("Resource {0} is assumed to be a directory.".format(directory_path))
                 artifact_path_list = [artifact.name for artifact in artifact_list]
                 self.link_manager.assign_listing(artifact_path_list)
             else:
-                content = storage.get_artifact(path)
-                self.link_manager.assign_single(content.key.name)
+                self.link_manager.assign_single(path)
 
-        return content
+            return artifact_list
```

Comment:
Can we outdent this and default artifact_list to an empty list?  I understand that if we can't connect to the bucket storage we will end up throwing an exception but i'd like to do this in case the code changes and so that it is easier (at least for me?) to understand.  If it is at the root level I can see that we always ensure we return an array